src/components/account/details.js

```diff
@@ -1,6 +1,101 @@
 // Libraries
-import React from 'react'
+import React, { Component } from 'react'
+import { Formik, Form, Field, ErrorMessage } from 'formik'
+import * as Yup from 'yup'
 
-export default () => (
-  <h1>Details</h1>
-)
+// Lib
+import componentMapping from '../../lib/component-mapping'
+
+class AccountDetails extends Component {
+  constructor (props) {
+    super(props)
+
+    this.Button = componentMapping('Button')
+    this.Flash = componentMapping('Flash')
+  }
+
+  renderFlash (status) {
+    switch (status) {
+      case 'success':
+        return (<this.Flash modifier='success' text='Your details were successfully updated.' />)
+      case 'error':
+        return (<this.Flash modifier='error' text='Oops, there was an error submitting your form.' />)
+    }
+  }
+
+  render () {
+    const { email, firstName, handleSubmit, lastName } = this.props
+
+    const validationSchema = Yup.object().shape({
+      firstName: Yup.string()
+        .required('Required'),
+      lastName: Yup.string()
+        .required('Required'),
+      email: Yup.string()
+        .email('Invalid email')
+        .required('Required'),
+      emailConfirmation: Yup.string()
+        .required('Required')
+        .oneOf([Yup.ref('email'), null], 'Must match')
+    })
+
+    return (
+      <Formik
+        enableReinitialize
+        initialValues={{
+          firstName: firstName || '',
+          lastName: lastName || '',
+          email: email || '',
+          emailConfirmation: ''
+        }}
+        validationSchema={validationSchema}
+        onSubmit={handleSubmit}
+        render={({ isSubmitting, isValid, status }) => {
+          const submitEnabled = isValid && !isSubmitting
+          return (
+            <>
+              {this.renderFlash(status)}
+              <Form>
```

Comment:
would be nice if these Fields and labels accepted props like `inputClassName`, `labelClassName` etc. Would make it much more customisable when it comes to using this component on a clients front-end.

eg:
```<Field className={classNames('o-form__input-field o-form__input-block', { [`inputClassName`]: inputClassName })}>```.

I'm planning on implementing this sort of customisation to many of our other components so would be good to get it straight in on these new ones.

Reply:
It is a very valid comment but I think it is best to get a one working example in the Trendy site first so that we can establish a convention. I'd like to avoid adding this kind of functionality everywhere now only to find out we need something _slightly_ different later.

Note that all these elements already have classes, you could potentially abandon the CSS for forms and implement your own.

I'll hold out until you start tackling customisation with @jameskirkby and see what solution we come up with.

Reply:
👍 no worries mate